nuxeo-features/nuxeo-platform-notification/nuxeo-platform-notification-core/src/test/java/org/nuxeo/ecm/platform/ec/notification/operations/SubscribeAndUnsubscribeTest.java

```diff
@@ -112,7 +112,7 @@ public void testSubscribeAndUnsubscribeOperations() throws OperationException {
 
         for (DocumentModel doc : listDocModel) {
             List<?> docSubscriptions = doc.getAdapter(SubscriptionAdapter.class).getUserSubscriptions(username);
-            assertEquals(4, docSubscriptions.size());
+            assertEquals(5, docSubscriptions.size());
```

Comment:
Does this means that the user is automatically subscribed to the `ColdStorageContentAvailable` notification?
Though, the attribute is set to `false` in the contribution.
So when is the user actually subscribed?

Reply:
> Does this means that the user is automatically subscribed to the ColdStorageContentAvailable notification?
No there is not auto-subscribing for this notification 

Here there is a call to the operation:
```
@Test
    public void testSubscribeAndUnsubscribeOperations() throws OperationException {
        OperationChain chain = new OperationChain("test-chain");
        chain.add(SubscribeOperation.ID);
        ctx.setInput(listDocModel);
        String username = NotificationConstants.USER_PREFIX + 
        ctx.getPrincipal().getName();
.....
``` 

and the  chain.add(SubscribeOperation.ID);  will subscribe all notifications available for a given document and now see [here](https://github.com/nuxeo/nuxeo/blob/master/nuxeo-features/nuxeo-platform-notification/nuxeo-platform-notification-core/src/main/java/org/nuxeo/ecm/platform/ec/notification/SubscriptionAdapter.java#L165)

> Though, the attribute is set to false in the contribution.
So when is the user actually subscribed?

yes i asked this question on our private slack channel, the PO need to subscribe automatically (without user action) when he asked for a retrieval. but i encounter the limitation of the fact that at `core`/`operation` level we don't have  access to the `core-notification` module which responsible of subscribing.

Reply:
We talked about adding a dependency to nuxeo-platform-notification-core in nuxeo-automation-core but this will induce a dependency cycle because nuxeo-automation-core depends on nuxeo-platform-notification-core...

In fact, the only reason for this is because of the `SubscribeOperation` and `UnsubscribeOperation`!

So I would suggest:
- You move these 2 operations from nuxeo-platform-notification-core to nuxeo-automation-features that already depends on nuxeo-platform-notification-core for the mail operations to it should be smooth.
- You remove the dependency to nuxeo-automation-core in nuxeo-platform-notification-core.
- You add a dependency to nuxeo-platform-notification-core in nuxeo-automation-core.
- You add the auto-subscribe code in the `RetrieveFromColdStorage` operation.

Reply:
yep, i will do / try it as soon as possible :) 
thanks

Reply:
> yep i completely agree that adding the `nuxeo-platform-notification-core` to `nuxeo-automation-core` will introduce a cyclic dependency because already the `nuxeo-platform-notification-core` import the `nuxeo-automation-core` dependency and yes this is due to `SubscribeOperation` and `UnsubscribeOperation` and yes they should be part of the `nuxeo-automation-features`. I think it's a good idea to move theses operation to `nuxeo-automation-features` and change the maven dependencies. i started tried it, for instant it works correctly moving the operation + there related tests... but i have some questions/interrogations:

- I am wondering if moving these operations will not break the backward compatibility as they are public we don't should remove them ? 

- I don't know if theses two operations can be referenced in a customer code i which case this will break changes 

i will continue the devs/checks and keep you updated

Reply:
It won't break anything since we are just moving them from one jar to another, that is always deployed in the server.

The only question is, do we need to keep the Java package `org.nuxeo.ecm.platform.ec.notification.automation` in case someone would extend these operations, which is usually not the case, or can we afford to change it to `org.nuxeo.ecm.automation.core.operations.notification` with its other operation friends.

Reply:
>It won't break anything since we are just moving them from one jar to another, that is always deployed in the server.

yes sorry i meant moving, but yep and as discuss theses changes will be internally, and if it is not inherited that will be fine and we can move it without breaking anything

Reply:
as discuss this once is done on: https://github.com/nuxeo/nuxeo/pull/3731